src/SDKs/Search/DataPlane/Search.Tests/Tests/IndexBatchExceptionTests.cs

```diff
@@ -103,10 +103,10 @@ private static IndexingResult FailedResult(string key, int statusCode)
         private void AssertRetryBatchContains(params string[] expectedKeys)
         {
             Assert.Equal(
-                expectedKeys, 
-                GetRetryBatch().Actions.Select(a => a.Document[KeyFieldName]).Cast<string>());
+                expectedKeys,
+                GetRetryBatch().Actions.Select(a => (a.Document as Document)[KeyFieldName]).Cast<string>());
```

Comment:
Both IndexBatch and IndexAction classes are now generated and since there's no way to describe generics in swagger and store types you will have to downcast the stored generic `object` for reading once it's in the IndexBatch. I understand your concern. However, I am not sure if it's strong enough to justify bringing back the removed custom generic classes (IndexBatch, IndexAction) in typed languages. In this specific case, if you aren't reading the content/keys, you do not need the downcast. The returned retry batch object can be passed into the Index operation without casting as before. I think we need to make a trade off. I will be happy to discuss.

Reply:
correction- if it's strong enough to justify bringing back the removed custom generic classes (IndexBatch, IndexAction) in typed languages _as well as in weakly typed languages._